core/src/env.rs

```diff
@@ -391,4 +391,7 @@ pub trait Externalities {
 
     /// Return the set of functions that are forbidden to be called.
     fn forbidden_funcs(&self) -> &BTreeSet<SyscallName>;
+
+    /// Return the set of functions that are forbidden to be called in this endpoint.
```

Comment:
I'd remove this. On backend side for each execution better to accept some "forbidden_funcs". Endpoint one's to be taken on backend from kind itself